code/game/mecha/combat/fighter.dm

```diff
@@ -222,6 +222,13 @@
 	else
 		who << sound('sound/mecha/fighter_entered.ogg',volume=50)
 
+//causes damage when running into objects
+/obj/mecha/combat/fighter/Bump(var/atom/obstacle)
```

Comment:
no var/, also needs ot . = ..()

also this check .. it WAS tested yes? this doesn't work on turfs. furthermore the damage is pretty high.

Reply:
it was taking damage when running into turfs as well as objects. dunno how, but it was.